instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/builder/HttpClientConfigBuilder.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.incubator.builder;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.instrumentation.api.incubator.semconv.http.HttpClientExperimentalMetrics;
+import io.opentelemetry.instrumentation.api.incubator.semconv.http.HttpClientPeerServiceAttributesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.http.HttpExperimentalAttributesExtractor;
+import io.opentelemetry.instrumentation.api.incubator.semconv.net.PeerServiceResolver;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpClientAttributesExtractorBuilder;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpClientMetrics;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpSpanNameExtractorBuilder;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpSpanStatusExtractor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+public abstract class HttpClientConfigBuilder<SELF, REQUEST, RESPONSE> {
```

Comment:
does it work to introduce and use this class throughout, without the other changes in this PR? if so, that would be an easy win (removing duplicate code and improving consistency) and we could tackle reviewing the harder parts after that?

Reply:
As far as I can see not.

This class uses the builder for the agent and spring starter, which both use config properties - which is why I started the PR:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c49f13c2b5deab24d51a4054c09dca2f0c01b7cb/sdk-autoconfigure-support/src/main/java/io/opentelemetry/instrumentation/internal/instrumenter/HttpClientInstrumenter.java

To make this possible I moved the config access to a different module:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c49f13c2b5deab24d51a4054c09dca2f0c01b7cb/sdk-autoconfigure-support/src/main/java/io/opentelemetry/instrumentation/internal/config/CoreCommonConfig.java

Reply:
I found an easier approach that leaves out the starter https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11620